diff --git a/config/aug_search.yml b/config/aug_search.yml
index 7feffe0..0f085b8 100644
--- a/config/aug_search.yml
+++ b/config/aug_search.yml
@@ -1,8 +1,8 @@
 mode: train
 seed: 128
-validation_frequency: 5
+validation_frequency: 2
 epochs: 50
-batch_size: 28
+batch_size: 42
 num_classes: 4
 train_dataset: 
   name: fgvc7
diff --git a/config/test_16.yml b/config/test_16.yml
index 9c96238..672027c 100644
--- a/config/test_16.yml
+++ b/config/test_16.yml
@@ -1,5 +1,5 @@
 mode: test
-seed: 128
+seed: 32
 test_dataset: 
   name: fgvc7
   resize_dims: 320
diff --git a/run.sh b/run.sh
index 16ea8d8..2f60f3c 100755
--- a/run.sh
+++ b/run.sh
@@ -67,4 +67,15 @@
 # python ./main.py -p 'exp_16_c.yml'
 # python ./main.py -p 'exp_16_d.yml'
 # python ./main.py -p 'exp_16_e.yml'
-python ./main.py 'test_16.yml'
\ No newline at end of file
+# python ./main.py 'test_16.yml'
+
+# python ./main.py -p 'exp_17.yml'
+
+# python ./main.py -p 'exp_18_a.yml'
+
+python ./main.py -p 'exp_19_a.yml'
+python ./main.py -p 'exp_19_b.yml'
+python ./main.py -p 'exp_19_c.yml'
+python ./main.py -p 'exp_19_d.yml'
+python ./main.py -p 'exp_19_e.yml'
+python ./main.py 'test_19.yml'
\ No newline at end of file
diff --git a/wandb/debug.log b/wandb/debug.log
index 61e8336..2c74182 100644
Binary files a/wandb/debug.log and b/wandb/debug.log differ
