diff --git a/run.sh b/run.sh
index 16ea8d8..af82dab 100755
--- a/run.sh
+++ b/run.sh
@@ -67,4 +67,6 @@
 # python ./main.py -p 'exp_16_c.yml'
 # python ./main.py -p 'exp_16_d.yml'
 # python ./main.py -p 'exp_16_e.yml'
-python ./main.py 'test_16.yml'
\ No newline at end of file
+# python ./main.py 'test_16.yml'
+
+python ./main.py -p 'exp_17.yml'
\ No newline at end of file
diff --git a/transformers/utils.py b/transformers/utils.py
index 517abc8..90e5487 100644
--- a/transformers/utils.py
+++ b/transformers/utils.py
@@ -127,7 +127,8 @@ class PolicyTransformer:
             if path.exists('transformers/best_policy.json'):
                 with open('transformers/best_policy.json') as f:
                     policy_array = json.load(f)
-                    sub_policy_pool = chain.from_iterable(policy_array)
+                    # sub_policy_pool = chain.from_iterable(policy_array)
+                    sub_policy_pool = policy_array[-1]
 
                     for sub_policy in sub_policy_pool:
                         op_1, params_1 = sub_policy[0]
diff --git a/wandb/debug.log b/wandb/debug.log
index 61e8336..8be9f42 100644
Binary files a/wandb/debug.log and b/wandb/debug.log differ
